cmd/mdatagen/validate_test.go

```diff
@@ -101,3 +105,38 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestMetricCollision(t *testing.T) {
+	allMetrics := map[string][]string{}
+	err := filepath.Walk("../../receiver", func(path string, info fs.FileInfo, err error) error {
+		if info.Name() == "metadata.yaml" {
+			// TODO: Remove once https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24011 is merged.
+			if path == "../../receiver/googlecloudspannerreceiver/internal/metadataconfig/metadata.yaml" {
+				return nil
+			}
+			md, err := loadMetadata(path)
+			assert.NoError(t, err)
+			if len(md.Metrics) > 0 {
+				for metricName := range md.Metrics {
+					allMetrics[md.Type] = append(allMetrics[md.Type], string(metricName))
+				}
+			}
+		}
+		return nil
+	})
+	assert.NoError(t, err)
+
+	seen := make(map[string]string)
+	for receiver, metrics := range allMetrics {
+		for _, metricName := range metrics {
+			// TODO: Remove one https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24009 and
+			// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24008 have been addressed.
+			if metricName == "container.cpu.utilization" || metricName == "container.memory.rss" {
+				continue
+			}
+			val, ok := seen[metricName]
+			assert.False(t, ok, fmt.Sprintf("Collision for metric %v in receivers %v and %v \n", metricName, receiver, val))
```

Comment:
Feel free to disregard, I understand that `ok` may be a bit of the default name here for whether a key exists in the map, but it seems counter-intuitive in this case. A test passing because something is not `ok` is just a bit hard to follow. 

Maybe `exists` or `previouslySeen`, something like that? Not a big deal though.

Reply:
Agreed, changed `ok` to `exists` ✔️